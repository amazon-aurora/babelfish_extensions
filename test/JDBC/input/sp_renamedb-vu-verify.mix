-- tsql
-- sanity checks for metadata stored in babelfish catalog
select owner, name from sys.babelfish_sysdatabases where name = 'rename_db_database1';
select nspname, orig_name from sys.babelfish_namespace_ext where nspname LIKE 'rename_db_database%';
select rolname, login_name, orig_username, database_name from sys.babelfish_authid_user_ext where rolname LIKE 'rename_db_database%';
select rolname, default_database_name from sys.babelfish_authid_login_ext where default_database_name LIKE 'rename_db_database%';
go

-- sanity check
use rename_db_database1;
go
-- Renaming the current database is also allowed
exec sp_renamedb 'rename_db_database1' , 'rename_db_database2';
go
select db_name()
go
use rename_db_database1;
go
use rename_db_database2;
go
exec sp_renamedb 'rename_db_database2', 'rename_db_database1';
go

use master
go
exec sp_renamedb 'rename_db_database1', 'rename_db_database2';
go
-- should return updated rows
select owner, name from sys.babelfish_sysdatabases where name LIKE 'rename_db_database%';
select nspname, orig_name from sys.babelfish_namespace_ext where nspname LIKE 'rename_db_database%';
select rolname, login_name, orig_username, database_name from sys.babelfish_authid_user_ext where rolname LIKE 'rename_db_database%';
select rolname, default_database_name from sys.babelfish_authid_login_ext where default_database_name LIKE 'rename_db_database%';
go

-- sanity check to use db
use rename_db_database1;
go
use rename_db_database2;
go

use master
go

-- rollback changes
exec sp_renamedb 'rename_db_database2', 'rename_db_database1'
go

-- sp_renamedb inside transaction is allowed
Begin Transaction;
go
exec sp_renamedb 'rename_db_database2', 'rename_db_database1'
go
-- does not rollback transaction
select @@trancount
go
commit
go

-- exec sp_renamedb inside a function/procedure should pass
create function sp_renamedb_function(@a text, @b text) returns int Begin exec sp_renamedb @a, @b END
go
create procedure sp_renamedb_procedure(@a text, @b text) as exec sp_renamedb @a, @b
go

select sp_renamedb_function('rename_db_database1', 'rename_db_database2');
go
exec sp_renamedb_procedure @a = 'rename_db_database1', @b = 'rename_db_database1'
go

-- Renaming system databases is prohibited
exec sp_renamedb master modify name rename_db_database1;
go
exec sp_renamedb tempdb modify name rename_db_database1;
go
exec sp_renamedb msdb modify name rename_db_database1;
go

-- Rename a case sensitive and len>64 database name
exec sp_renamedb '[ThisOldDatabaseNameIsCaseSensitiveAndIsLongerThan64DigitsToTestRenameDb]',
 '[ThisNewDatabaseNameIsCaseSensitiveAndIsLongerThan64DigitsToTestRenameDb]'
go

-- should return updated rows
select owner, name from sys.babelfish_sysdatabases where name LIKE 'DatabaseNameIsCaseSensitiveAndIsLongerThan64DigitsToTestRenameDb%';
select nspname, orig_name from sys.babelfish_namespace_ext where nspname LIKE 'DatabaseNameIsCaseSensitiveAndIsLongerThan64DigitsToTestRenameDb%';
select rolname, login_name, orig_username, database_name from sys.babelfish_authid_user_ext where rolname LIKE 'DatabaseNameIsCaseSensitiveAndIsLongerThan64DigitsToTestRenameDb%';
select rolname, default_database_name from sys.babelfish_authid_login_ext where default_database_name LIKE 'DatabaseNameIsCaseSensitiveAndIsLongerThan64DigitsToTestRenameDb%';
go

-- sanity check to use db
use [ThisOldDatabaseNameIsCaseSensitiveAndIsLongerThan64DigitsToTestRenameDb];
go
use [ThisNewDatabaseNameIsCaseSensitiveAndIsLongerThan64DigitsToTestRenameDb]
go

use master
go

-- tsql user=rename_db_login1 password=1234
use rename_db_database1
go
-- user that does not have access to db, cannot modify name
exec sp_renamedb 'rename_db_database1', 'rename_db_database2';
go

-- tsql
-- give rename_db_login1 permission to the db
alter server role sysadmin add member rename_db_login1
go

-- tsql user=rename_db_login1 password=1234
-- should work now
exec sp_renamedb 'rename_db_database1', 'rename_db_database2';
go
-- connect to the database to test concurrencies
use rename_db_database2
go

-- tsql
-- should throw error since there is an active session using rename_db_database2
exec sp_renamedb 'rename_db_database2', 'rename_db_database1';
go

-- tsql user=rename_db_login1 password=1234
-- now disconnect from that database in the 2nd connection and test alter again
select db_name()
go
use master
go

-- tsql
exec sp_renamedb 'rename_db_database2', 'rename_db_database1';
go

drop function sp_renamedb_function
drop procedure sp_renamedb_procedure
go