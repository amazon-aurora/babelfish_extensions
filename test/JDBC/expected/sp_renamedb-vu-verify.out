-- tsql
-- sanity checks for metadata stored in babelfish catalog
select owner, name from sys.babelfish_sysdatabases where name = 'rename_db_database1';
select nspname, orig_name from sys.babelfish_namespace_ext where nspname LIKE 'rename_db_database%';
select rolname, login_name, orig_username, database_name from sys.babelfish_authid_user_ext where rolname LIKE 'rename_db_database%';
select rolname, default_database_name from sys.babelfish_authid_login_ext where default_database_name LIKE 'rename_db_database%';
go
~~START~~
varchar#!#text
kush#!#rename_db_database1
~~END~~

~~START~~
varchar#!#nvarchar
rename_db_database1_dbo#!#dbo
rename_db_database1_guest#!#guest
rename_db_database1_rename_db_schema1#!#rename_db_schema1
~~END~~

~~START~~
varchar#!#varchar#!#nvarchar#!#nvarchar
rename_db_database1_dbo#!##!#dbo#!#rename_db_database1
rename_db_database1_db_owner#!##!#db_owner#!#rename_db_database1
rename_db_database1_guest#!##!#guest#!#rename_db_database1
rename_db_database1_rename_db_role1#!##!#rename_db_role1#!#rename_db_database1
rename_db_database1_rename_db_login2#!#rename_db_login2#!#rename_db_login2#!#rename_db_database1
~~END~~

~~START~~
varchar#!#nvarchar
rename_db_login1#!#rename_db_database1
~~END~~


-- sanity check
use rename_db_database1;
go
-- Renaming the current database is also allowed
exec sp_renamedb 'rename_db_database1' , 'rename_db_database2';
go
select db_name()
go
~~START~~
nvarchar
rename_db_database2
~~END~~

use rename_db_database1;
go
~~ERROR (Code: 911)~~

~~ERROR (Message: database "rename_db_database1" does not exist)~~

use rename_db_database2;
go
exec sp_renamedb 'rename_db_database2', 'rename_db_database1';
go

use master
go
exec sp_renamedb 'rename_db_database1', 'rename_db_database2';
go
-- should return updated rows
select owner, name from sys.babelfish_sysdatabases where name LIKE 'rename_db_database%';
select nspname, orig_name from sys.babelfish_namespace_ext where nspname LIKE 'rename_db_database%';
select rolname, login_name, orig_username, database_name from sys.babelfish_authid_user_ext where rolname LIKE 'rename_db_database%';
select rolname, default_database_name from sys.babelfish_authid_login_ext where default_database_name LIKE 'rename_db_database%';
go
~~START~~
varchar#!#text
kush#!#rename_db_database2
~~END~~

~~START~~
varchar#!#nvarchar
rename_db_database2_dbo#!#dbo
rename_db_database2_guest#!#guest
rename_db_database2_rename_db_schema1#!#rename_db_schema1
~~END~~

~~START~~
varchar#!#varchar#!#nvarchar#!#nvarchar
rename_db_database2_dbo#!##!#dbo#!#rename_db_database2
rename_db_database2_db_owner#!##!#db_owner#!#rename_db_database2
rename_db_database2_guest#!##!#guest#!#rename_db_database2
rename_db_database2_rename_db_role1#!##!#rename_db_role1#!#rename_db_database2
rename_db_database2_rename_db_login2#!#rename_db_login2#!#rename_db_login2#!#rename_db_database2
~~END~~

~~START~~
varchar#!#nvarchar
rename_db_login1#!#rename_db_database2
~~END~~


-- sanity check to use db
use rename_db_database1;
go
~~ERROR (Code: 911)~~

~~ERROR (Message: database "rename_db_database1" does not exist)~~

use rename_db_database2;
go

use master
go

-- rollback changes
exec sp_renamedb 'rename_db_database2', 'rename_db_database1'
go

-- sp_renamedb inside transaction is allowed
Begin Transaction;
go
exec sp_renamedb 'rename_db_database2', 'rename_db_database1'
go
~~ERROR (Code: 574)~~

~~ERROR (Message: SP_RENAME/SP_RENAMEDB cannot run inside a transaction block)~~

-- does not rollback transaction
select @@trancount
go
~~START~~
int
1
~~END~~

commit
go

-- exec sp_renamedb inside a function/procedure should pass
create function sp_renamedb_function(@a text, @b text) returns int Begin exec sp_renamedb @a, @b END
go
create procedure sp_renamedb_procedure(@a text, @b text) as exec sp_renamedb @a, @b
go

select sp_renamedb_function('rename_db_database1', 'rename_db_database2');
go
~~START~~
int
~~ERROR (Code: 33557097)~~

~~ERROR (Message: Only functions can be executed within a function)~~

exec sp_renamedb_procedure @a = 'rename_db_database1', @b = 'rename_db_database1'
go
~~ERROR (Code: 33557097)~~

~~ERROR (Message: schema "rename_db_database1_dbo" already exists)~~


-- Renaming system databases is prohibited
exec sp_renamedb master modify name rename_db_database1;
go
~~ERROR (Code: 33557097)~~

~~ERROR (Message: syntax error near 'name' at line 2 and character position 31)~~

exec sp_renamedb tempdb modify name rename_db_database1;
go
~~ERROR (Code: 33557097)~~

~~ERROR (Message: syntax error near 'name' at line 1 and character position 31)~~

exec sp_renamedb msdb modify name rename_db_database1;
go
~~ERROR (Code: 33557097)~~

~~ERROR (Message: syntax error near 'name' at line 1 and character position 29)~~


-- Rename a case sensitive and len>64 database name
exec sp_renamedb '[ThisOldDatabaseNameIsCaseSensitiveAndIsLongerThan64DigitsToTestRenameDb]',
 '[ThisNewDatabaseNameIsCaseSensitiveAndIsLongerThan64DigitsToTestRenameDb]'
go

-- should return updated rows
select owner, name from sys.babelfish_sysdatabases where name LIKE 'DatabaseNameIsCaseSensitiveAndIsLongerThan64DigitsToTestRenameDb%';
select nspname, orig_name from sys.babelfish_namespace_ext where nspname LIKE 'DatabaseNameIsCaseSensitiveAndIsLongerThan64DigitsToTestRenameDb%';
select rolname, login_name, orig_username, database_name from sys.babelfish_authid_user_ext where rolname LIKE 'DatabaseNameIsCaseSensitiveAndIsLongerThan64DigitsToTestRenameDb%';
select rolname, default_database_name from sys.babelfish_authid_login_ext where default_database_name LIKE 'DatabaseNameIsCaseSensitiveAndIsLongerThan64DigitsToTestRenameDb%';
go
~~START~~
varchar#!#text
~~END~~

~~START~~
varchar#!#nvarchar
~~END~~

~~START~~
varchar#!#varchar#!#nvarchar#!#nvarchar
~~END~~

~~START~~
varchar#!#nvarchar
~~END~~


-- sanity check to use db
use [ThisOldDatabaseNameIsCaseSensitiveAndIsLongerThan64DigitsToTestRenameDb];
go
~~ERROR (Code: 911)~~

~~ERROR (Message: database "thisolddatabasenameiscasesensit70deaa61d81bc286d3760d346a37224c" does not exist)~~

use [ThisNewDatabaseNameIsCaseSensitiveAndIsLongerThan64DigitsToTestRenameDb]
go

use master
go

-- tsql user=rename_db_login1 password=1234
use rename_db_database1
go
~~ERROR (Code: 33557097)~~

~~ERROR (Message: The server principal "rename_db_login1" is not able to access the database "rename_db_database1" under the current security context)~~

-- user that does not have access to db, cannot modify name
exec sp_renamedb 'rename_db_database1', 'rename_db_database2';
go
~~ERROR (Code: 33557097)~~

~~ERROR (Message: Current login rename_db_login1 does not have permission to create new login)~~


-- tsql
-- give rename_db_login1 permission to the db
alter server role sysadmin add member rename_db_login1
go

-- tsql user=rename_db_login1 password=1234
-- should work now
exec sp_renamedb 'rename_db_database1', 'rename_db_database2';
go
-- connect to the database to test concurrencies
use rename_db_database2
go

-- tsql
-- should throw error since there is an active session using rename_db_database2
exec sp_renamedb 'rename_db_database2', 'rename_db_database1';
go
~~ERROR (Code: 33557097)~~

~~ERROR (Message: The database could not be exclusively locked to perform the operation.)~~


-- tsql user=rename_db_login1 password=1234
-- now disconnect from that database in the 2nd connection and test alter again
select db_name()
go
~~START~~
nvarchar
rename_db_database2
~~END~~

use master
go

-- tsql
exec sp_renamedb 'rename_db_database2', 'rename_db_database1';
go

drop function sp_renamedb_function
drop procedure sp_renamedb_procedure
go
